sale_double_validation/models/sale.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    state = fields.Selection(
+        selection_add=[('to_approve', 'To Approve')]
+    )
+
+    @api.multi
+    def is_amount_to_approve(self):
+        currency = self.env.user.company_id.currency_id
+        limit_amount = self.company_id.so_double_validation_amount
+        limit_amount = currency.compute(limit_amount, self.currency_id)
+        return self.amount_total >= limit_amount
+
+    @api.multi
+    def is_to_approve(self):
+        return (self.company_id.so_double_validation == 'two_step' and
+                self.is_amount_to_approve() and
+                not self.user_has_groups('sales_team.group_sale_manager'))
+
+    @api.multi
+    def action_confirm(self):
+        to_approve = self.env['sale.order']
+        to_confirm = self.env['sale.order']
```

Comment:
This is really confusing and looks like an error on fast reading of the code.
Even if under the cover it has the same effect, I suggest the following lines. I think they better show the intent which is to create new recordset containers for gathering records:

```
to_approve = self.env['sale.order'].browse()
to_confirm = self.env['sale.order'].browse()
```